robotoff/prediction/ocr/category_from_AOC.py

```diff
@@ -0,0 +1,84 @@
+import re
+from typing import Dict, Iterable, List, Optional, Union
+
+from robotoff import settings
+from robotoff.prediction.types import Prediction, PredictionType
+from robotoff.taxonomy import Taxonomy, TaxonomyNode, get_taxonomy
+from robotoff.utils import get_logger
+
+from .dataclass import OCRField, OCRRegex, OCRResult, get_text
+
+logger = get_logger(__name__)
+
+
+def process_category(lang, match) -> Optional[str]:
+    """Function to process the name of the category extracted from the AOC
+    recognition. We want it to match categories found in the categories
+    taxonomy database. If nothing is found, we don't return the category."""
+
+    unchecked_category = str(lang) + ":" + match.group().replace(" ", "-")
+
+    checked_category = get_taxonomy("category").__getitem__(unchecked_category)
+
+    if checked_category is not None:
+        return unchecked_category
+
+    return None
```

Comment:
I agree it does look a bit stupid. 
But actually checked_category is a Taxonomy and we must return a string so I can't return checked_category itself.
That's why I say : 
- if our unchecked_category detected by AOP REGEX does exist among already existing categories, we return it
- else we can't return unchecked_category because it doesn't match an existing category, so we return None

Reply:
oooh that's me, the stupid !
In didn't saw **un**checked_category

Your code was correct.

That said you should return the checked_category tag because in find_category_from_AOC you have to put the tag for this category not the variable string you found.